src/controllers/company.details.controller.ts

```diff
@@ -70,6 +79,14 @@ export const confirmCompanyStartRequest = async (req: Request, res: Response, ne
           `for companyNumber ${companyNumber}`);
       }
       await sessionService.updateExtensionSessionValue(req.chSession, keys.ALREADY_SUBMITTED, false);
+
+      // check they haven't extended too far previously
+      //   - need the extension request to exist before updating its status
+      if (!isExtensionDueDateWithinLimit(company)) {
+        await setExtensionStatusToRejected(chSession, companyNumber, token);
+        return res.redirect(pageURLs.EXTENSIONS_EXTENSION_LIMIT_REACHED);
```

Comment:
That logger.info statement that was there previously could be useful. Would suggest re-including it.